src/duiker.py

```diff
@@ -36,7 +36,7 @@ def xdg_data_home(name: Optional[str] = None) -> pathlib.Path:
     Return the XDG Base Directory Specification data directory for a specific
     application, or the base directory itself.
     """
-    home = pathlib.Path(os.path.expanduser(os.environ.get('XDG_DATA_HOME') or '~/.local/share'))
+    home = pathlib.Path(os.path.expanduser(os.environ.get('XDG_DATA_HOME') or os.path.expanduser('~') + '/.local/share'))
```

Comment:
Now I can't reproduce using your master. It was odd that it was reporting a write permissions problem on ~, but even after making sure with `chmod -R` it would not resolve. The only way I could get it to work on CentOS 7.3 was the change I made. I am testing on the same box, don't have a fresh one handy. This was a terrible PR - apologies.